apps/zipsync/src/zipUtils.ts

```diff
@@ -0,0 +1,337 @@
+  commentLength: number;
+}
+
+function writeUInt32LE(buffer: Buffer, value: number, offset: number): void {
+  buffer.writeUInt32LE(value, offset);
+}
+
+function writeUInt16LE(buffer: Buffer, value: number, offset: number): void {
+  buffer.writeUInt16LE(value, offset);
+}
+
+function readUInt32LE(buffer: Buffer, offset: number): number {
+  return buffer.readUInt32LE(offset);
+}
+
+function readUInt16LE(buffer: Buffer, offset: number): number {
+  return buffer.readUInt16LE(offset);
+}
+
+export function dosDateTime(date: Date): { time: number; date: number } {
+  /* eslint-disable no-bitwise */
+  const time: number =
+    ((date.getHours() & 0x1f) << 11) | ((date.getMinutes() & 0x3f) << 5) | ((date.getSeconds() / 2) & 0x1f);
+
+  const dateVal: number =
+    (((date.getFullYear() - 1980) & 0x7f) << 9) |
+    (((date.getMonth() + 1) & 0xf) << 5) |
+    (date.getDate() & 0x1f);
+  /* eslint-enable no-bitwise */
+
+  return { time, date: dateVal };
+}
+
+const localFileHeaderBuffer: Buffer = Buffer.allocUnsafe(30);
+export function writeLocalFileHeader(
+  entry: IFileEntry
+): [fileHeaderWithoutVariableLengthData: Buffer, fileHeaderVariableLengthData: Buffer] {
+  const filenameBuffer: Buffer = Buffer.from(entry.filename, 'utf8');
+
+  const { time, date } = entry.dosDateTime;
+
+  let offset: number = 0;
+  writeUInt32LE(localFileHeaderBuffer, LOCAL_FILE_HEADER_SIGNATURE, offset);
+  offset += 4;
+  writeUInt16LE(localFileHeaderBuffer, 20, offset); // version needed
+  offset += 2;
+  // General purpose bit flag: set bit 3 (0x0008) to indicate presence of data descriptor
+  // Per APPNOTE: when bit 3 is set, CRC-32 and sizes in local header are set to zero and
+  // the actual values are stored in the data descriptor that follows the file data.
+  writeUInt16LE(localFileHeaderBuffer, 0x0008, offset); // flags (data descriptor)
+  offset += 2;
+  writeUInt16LE(localFileHeaderBuffer, entry.compressionMethod, offset); // compression method (0=store,8=deflate)
+  offset += 2;
+  writeUInt16LE(localFileHeaderBuffer, time, offset); // last mod time
+  offset += 2;
+  writeUInt16LE(localFileHeaderBuffer, date, offset); // last mod date
+  offset += 2;
+  // With bit 3 set, these three fields MUST be zero in the local header
+  writeUInt32LE(localFileHeaderBuffer, 0, offset); // crc32 (placeholder, real value in data descriptor)
+  offset += 4;
+  writeUInt32LE(localFileHeaderBuffer, 0, offset); // compressed size (placeholder)
+  offset += 4;
+  writeUInt32LE(localFileHeaderBuffer, 0, offset); // uncompressed size (placeholder)
+  offset += 4;
+  writeUInt16LE(localFileHeaderBuffer, filenameBuffer.length, offset); // filename length
+  offset += 2;
+  writeUInt16LE(localFileHeaderBuffer, 0, offset); // extra field length
+  offset += 2;
+
+  return [localFileHeaderBuffer, filenameBuffer];
+}
+
+const centralDirHeaderBuffer: Buffer = Buffer.allocUnsafe(46);
+export function writeCentralDirectoryHeader(entry: IFileEntry): Buffer[] {
```

Comment:
Can you add some block comments for these detailing the binary layout of the section?